File: scripts/generate-og-image.py
#!/usr/bin/env python3
from PIL import Image, ImageDraw, ImageFont
import os

# Crear imagen de 1200x630 (tamaño recomendado para Open Graph)
width = 1200
height = 630
img = Image.new('RGB', (width, height), color='#0f172a')
draw = ImageDraw.Draw(img)

# Borde superior con color primary
draw.rectangle([0, 0, width, 8], fill='#f59e0b')

# Intentar cargar fuentes del sistema, si no usar default
try:
    font_title = ImageFont.truetype('/usr/share/fonts/truetype/dejavu/DejaVuSans-Bold.ttf', 100)
    font_subtitle = ImageFont.truetype('/usr/share/fonts/truetype/dejavu/DejaVuSans.ttf', 28)
    font_desc = ImageFont.truetype('/usr/share/fonts/truetype/dejavu/DejaVuSans.ttf', 24)
    font_url = ImageFont.truetype('/usr/share/fonts/truetype/dejavu/DejaVuSansMono.ttf', 20)
except:
    font_title = ImageFont.load_default()
    font_subtitle = ImageFont.load_default()
    font_desc = ImageFont.load_default()
    font_url = ImageFont.load_default()

# Título "PAPERBOY"
text = "PAPERBOY"
bbox = draw.textbbox((0, 0), text, font=font_title)
text_width = bbox[2] - bbox[0]
x = (width - text_width) / 2
draw.text((x, 150), text, fill='#ffffff', font=font_title)

# Subtítulo "LA PORTERÍA"
text = "LA PORTERÍA"
bbox = draw.textbbox((0, 0), text, font=font_subtitle)
text_width = bbox[2] - bbox[0]
x = (width - text_width) / 2
draw.text((x, 280), text, fill='#94a3b8', font=font_subtitle)

# Descripción línea 1
text = "Tiras cómicas diarias"
bbox = draw.textbbox((0, 0), text, font=font_desc)
text_width = bbox[2] - bbox[0]
x = (width - text_width) / 2
draw.text((x, 360), text, fill='#cbd5e1', font=font_desc)

# Descripción línea 2
text = "con estilo minimalista tipo periódico"
bbox = draw.textbbox((0, 0), text, font=font_desc)
text_width = bbox[2] - bbox[0]
x = (width - text_width) / 2
draw.text((x, 400), text, fill='#cbd5e1', font=font_desc)

# Línea decorativa
line_y = 460
draw.line([(width/2 - 200, line_y), (width/2 + 200, line_y)], fill='#f59e0b', width=3)

# URL
text = "albertomaydayjhondoe.github.io/Porterias"
bbox = draw.textbbox((0, 0), text, font=font_url)
text_width = bbox[2] - bbox[0]
x = (width - text_width) / 2
draw.text((x, 530), text, fill='#64748b', font=font_url)

# Guardar la imagen
output_path = os.path.join(os.path.dirname(__file__), '..', 'public', 'og-image.png')
img.save(output_path, 'PNG')
print(f'✓ Imagen Open Graph generada: {output_path}')

File: scripts/generate-og-image.js
import { createCanvas, loadImage } from 'canvas';
import fs from 'fs';
import path from 'path';
import { fileURLToPath } from 'url';

const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);

async function generateOGImage() {
  // Crear canvas de 1200x630 (tamaño recomendado para Open Graph)
  const width = 1200;
  const height = 630;
  const canvas = createCanvas(width, height);
  const ctx = canvas.getContext('2d');

  // Fondo oscuro (color del tema)
  ctx.fillStyle = '#0f172a';
  ctx.fillRect(0, 0, width, height);

  // Borde superior con color primary
  ctx.fillStyle = '#f59e0b';
  ctx.fillRect(0, 0, width, 8);

  // Título "PAPERBOY"
  ctx.fillStyle = '#ffffff';
  ctx.font = 'bold 120px sans-serif';
  ctx.textAlign = 'center';
  ctx.fillText('PAPERBOY', width / 2, 200);

  // Subtítulo "La Portería"
  ctx.fillStyle = '#94a3b8';
  ctx.font = '32px sans-serif';
  ctx.letterSpacing = '0.3em';
  ctx.fillText('LA PORTERÍA', width / 2, 260);

  // Descripción
  ctx.fillStyle = '#cbd5e1';
  ctx.font = '28px sans-serif';
  ctx.fillText('Tiras cómicas diarias', width / 2, 360);
  ctx.fillText('con estilo minimalista tipo periódico', width / 2, 410);

  // Línea decorativa
  ctx.strokeStyle = '#f59e0b';
  ctx.lineWidth = 3;
  ctx.beginPath();
  ctx.moveTo(width / 2 - 200, 460);
  ctx.lineTo(width / 2 + 200, 460);
  ctx.stroke();

  // URL
  ctx.fillStyle = '#64748b';
  ctx.font = '24px monospace';
  ctx.fillText('albertomaydayjhondoe.github.io/Porterias', width / 2, 540);

  // Guardar la imagen
  const buffer = canvas.toBuffer('image/png');
  const outputPath = path.join(__dirname, '..', 'public', 'og-image.png');
  fs.writeFileSync(outputPath, buffer);
  console.log('✓ Imagen Open Graph generada:', outputPath);
}

generateOGImage().catch(console.error);
